web3/utils/abi.py

```diff
@@ -1,21 +1,242 @@
+import functools
 from typing import (
+    Any,
     List,
+    Optional,
+    Sequence,
     Union,
+    cast,
 )
 
+from eth_typing import (
+    ABIFunctionInfo,
+)
+from eth_utils.abi import (
+    function_abi_to_4byte_selector,
+    get_aligned_abi_inputs,
+    get_normalized_abi_inputs,
+)
+from eth_utils.encoding import (
+    get_default_codec,
+)
+from eth_utils.hexadecimal import (
+    encode_hex,
+)
+from eth_utils.toolz import (
+    pipe,
+)
+from eth_utils.types import (
+    is_text,
+)
+from typing_extensions import (
+    deprecated,
+)
+
+from web3._utils.abi import (
+    abi_to_signature,
+    filter_by_argument_count,
+    filter_by_encodability,
+    filter_by_name,
+)
+from web3._utils.contracts import (
+    extract_argument_types,
+)
+from web3.exceptions import (
+    MismatchedABI,
+)
 from web3.types import (
+    ABI,
     ABIEvent,
     ABIFunction,
 )
 
+DEPRECATION_MESSAGE = (
+    "`{0}` is deprecated and will be removed in a future release. Use " "`{1}` instead."
+)
+
 
+@deprecated(
+    DEPRECATION_MESSAGE.format(
+        "get_abi_input_names", "eth_utils.abi.get_abi_output_names"
+    )
+)
+@deprecated("Use eth_utils.abi.get_abi_input_names")
 def get_abi_input_names(abi: Union[ABIFunction, ABIEvent]) -> List[str]:
+    """
+    Return the ``input`` names for an ABI function or event.
+    """
     if "inputs" not in abi and abi["type"] == "fallback":
         return []
     return [arg["name"] for arg in abi["inputs"]]
 
 
+@deprecated(
+    DEPRECATION_MESSAGE.format(
+        "get_abi_output_names", "eth_utils.abi.get_abi_output_names"
+    )
+)
```

Comment:
Same here. Get rid of this implementation entirely for `v7` and just import it from *eth_utils* to re-expose?